samsung-cnct/kraken-lib
# [BAC-102] Add kube bench tests to CI (#660)

**What this PR does / why we need it**: This PR adds a CI job that runs the [kube-bench][2] tool on a Kraken cluster. The kube-bench tool is an implementation of the [CIS Benchmark for Kubernetes][1]. This will allow us to establish a baseline metric in regards to the security posture of a Kraken provisioned cluster.

**Which issue this PR fixes**: fixes #660 ([BAC-102][102])

**How to Verify**
- This PR should invoke the new `test-security:benchmark` (and `benchmark-minus-one` and `benchmark-minus-two`) in the gitlab pipeline. On success of this job the test result artifacts should be associated with the gitlab job. You should be able to verify these files manually.
- Additionally this can be invoked locally with the following command:
```
docker run \
  -v ~/.ssh:/root/.ssh \
  -v ~/.ssh:$(dirname ~/.ssh)/.ssh \
  -v ~/.kraken:/root/.kraken \
  -v $(PWD):/root/kraken-lib \
  -w /root/kraken-lib \
  quay.io/samsung_cnct/kraken-tools:latest  ./build-scripts/benchmark-tests "/root/.kraken"
  ```
By default this will try and run the bench mark on the cluster specified in `~/.kraken/config.yaml`

**Notes**:
- This issue does not address wether or not the recommended remediation of failed checks are valid for a Kraken cluster. A separate issue should be created to address this.
- A report for each node in the cluster will generated to `./builds/tests/`

[1]: https://www.cisecurity.org/benchmark/kubernetes/
[2]: https://github.com/aquasecurity/kube-bench
[102]: https://samsung-cnct.atlassian.net/browse/BAC-102
